clang/test/Driver/ms-define-stdc.c

```diff
@@ -0,0 +1,11 @@
+// Note: %s must be preceded by --, otherwise it may be interpreted as a
+// command-line option, e.g. on Mac where %s is commonly under /Users.
+//
+// Note: see also cl-zc.cpp
+
+// RUN: %clang_cl /TC /dev/null /E -Xclang -dM /Zc:__STDC__- 2>&1 | FileCheck %s --check-prefix=ZCSTDCIGNORED
+// ZCSTDCIGNORED-NOT: #define __STDC__ 1
+// ZCSTDCIGNORED: argument unused during compilation
+
+// RUN: not %clang -Xclang -fno-ms-define-stdc %s 2>&1 | FileCheck %s --check-prefix="NOARG"
```

Comment:
This tests that -fno-ms-define-stdc is not a cc1 option. We almost never do it. There could be a value to test whether the driver -fno-ms-define-stdc option is present.

Reply:
Hmmm, this test was requested by @AaronBallman - what would you do in this case?

Reply:
@MaskRay -- see https://github.com/llvm/llvm-project/pull/68690#discussion_r1355620210 as to why I think it's valuable to have this test (but if you think it's not a useful test, I don't insist on it).

Reply:
Sorry for the late reply. If you want to show the behavior of the driver option -fno-ms-define-stdc, feel free to add it when you think there is sufficient value.

For CC1 options, I believe we perform a negative test: hey, we check that the negative form is not present.
The value testing that seems low.